cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/customaction/PipelineAction.java

```diff
@@ -70,7 +70,13 @@ public void run() throws Exception {
     BatchPhaseSpec phaseSpec = GSON.fromJson(properties.get(Constants.PIPELINEID), BatchPhaseSpec.class);
     PipelinePhase phase = phaseSpec.getPhase();
     StageInfo stageInfo = phase.iterator().next();
-    Action action = context.newPluginInstance(stageInfo.getName());
+    Action action =
+      context.newPluginInstance(stageInfo.getName(),
+                                new DefaultMacroEvaluator(context.getWorkflowToken(),
+                                                          context.getRuntimeArguments(),
+                                                          context.getLogicalStartTime(),
+                                                          context,
```

Comment:
My suggested approach (adding the method to AbstractContext) wouldn't work, since the context isn't visible as an `AbstractContext`, but as a `CustomActionContext`, which wouldn't have the `getSecureStore` method, without changing the public facing API.